src/Uno.UWP/Graphics/Display/DisplayInformation.Android.cs

```diff
@@ -1,22 +1,184 @@
+					isLandscape = orientation == Android.Content.Res.Orientation.Portrait;
+					break;
+			}
+			NativeOrientation = isLandscape ? DisplayOrientations.Landscape : DisplayOrientations.Portrait;
+		}
+
+		/// <summary>
+		/// Sets the CurrentOrientation property
+		/// </summary>
+		/// <param name="windowManager">Window manager</param>
+		/// <remarks>
+		/// Again quite complicated to do on Android, based on accepted solution at
+		/// <see cref="https://stackoverflow.com/questions/10380989/how-do-i-get-the-current-orientation-activityinfo-screen-orientation-of-an-a">this SO question</see>
+		/// </remarks>
+		private void UpdateCurrentOrientation(IWindowManager windowManager)
+		{
+			var rotation = windowManager.DefaultDisplay.Rotation;
+			var displayMetrics = new DisplayMetrics();
+			windowManager.DefaultDisplay.GetMetrics(displayMetrics);
+
+			int width = displayMetrics.WidthPixels;
+			int height = displayMetrics.HeightPixels;
+
+			if ( width == height)
+			{
+				//square device, can't tell orientation
+				CurrentOrientation = DisplayOrientations.None;
+				return;
+			}
+			
+			var portraitOrientation0Or180 =
+				(rotation == SurfaceOrientation.Rotation0 ||
+				rotation == SurfaceOrientation.Rotation180)
+				&& height > width;
+			var portraitOrientation90Or270 =
+				(rotation == SurfaceOrientation.Rotation90 ||
+				rotation == SurfaceOrientation.Rotation270)
+				&& width > height;
+
+			// if the device's natural orientation is portrait:
+			if (
+				portraitOrientation0Or180 ||
+				portraitOrientation90Or270 )
+			{
+				switch (rotation)
+				{
+					case SurfaceOrientation.Rotation0:
+						CurrentOrientation = DisplayOrientations.Portrait;
+						break;
+					case SurfaceOrientation.Rotation90:
+						CurrentOrientation = DisplayOrientations.Landscape;
+						break;
+					case SurfaceOrientation.Rotation180:
+						CurrentOrientation = DisplayOrientations.PortraitFlipped;
+						break;
+					case SurfaceOrientation.Rotation270:
+						CurrentOrientation = DisplayOrientations.LandscapeFlipped;
+						break;
+					default:
+						//invalid rotation
+						CurrentOrientation = DisplayOrientations.None;
+						break;
+				}
+			}			
+			else 
+			{
+				//device is landscape
+				switch (rotation)
+				{
+					case SurfaceOrientation.Rotation0:
+						CurrentOrientation = DisplayOrientations.Landscape;
+						break;
+					case SurfaceOrientation.Rotation90:
+						CurrentOrientation = DisplayOrientations.Portrait;
+						break;
+					case SurfaceOrientation.Rotation180:
+						CurrentOrientation = DisplayOrientations.LandscapeFlipped;
+						break;
+					case SurfaceOrientation.Rotation270:
+						CurrentOrientation = DisplayOrientations.PortraitFlipped;
+						break;
+					default:
+						//invalid rotation
+						CurrentOrientation = DisplayOrientations.None;
+						break;
+				}
+			}
+		}
+
+
+		internal void HandleConfigurationChange()
+		{
+			var previousOrientation = CurrentOrientation;
+			Initialize();			
```

Comment:
Presumably only CurrentOrientation needs to be updated every time?

Reply:
No, `HandleConfigurationChange` can be triggered by `SizeChange` on Android, which can be something like switching to a completely new display, etc. Hence any of the properties could potentially change